python/cudf/cudf/core/column/datetime.py

```diff
@@ -82,10 +82,18 @@ def from_numpy(cls, array):
                 ("Cannot infer datetime dtype " + "from np.array dtype `%s`")
                 % (array.dtype)
             )
+        array_buffer = Buffer(array)
+        mask, null_count = libcudf.unaryops.nats_to_nulls(array_buffer)
```

Comment:
You need to hook this up into the `cudf.core.column.column.as_column` function as well.

Reply:
@kkraus14 Did you mean to hook this onto `pyarrow` section of as_column ?

Reply:
No, I mean here: https://github.com/rapidsai/cudf/blob/branch-0.10/python/cudf/cudf/core/column/column.py#L1188-L1196